src/OrchardCore/OrchardCore.ContentManagement.GraphQL/Mutations/DeleteContentItemMutation.cs

```diff
@@ -0,0 +1,58 @@
+using System.Threading.Tasks;
+using GraphQL.Resolvers;
+using GraphQL.Types;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.DependencyInjection;
+using OrchardCore.Apis.GraphQL;
+using OrchardCore.Apis.GraphQL.Types;
+using OrchardCore.ContentManagement;
+
+namespace OrchardCore.ContentManagement.GraphQL.Mutations
+{
+    public class DeleteContentItemMutation : MutationFieldType
+    {
+        private readonly IHttpContextAccessor _httpContextAccessor;
+
+        public DeleteContentItemMutation(IHttpContextAccessor httpContextAccessor)
+        {
+            Name = "DeleteContentItem";
+
+            Arguments = new QueryArguments(
+                new QueryArgument<NonNullGraphType<StringGraphType>> { Name = "contentItemId" }
+            );
+
+            Type = typeof(DeletionStatusObjectGraphType);
+
+            Resolver = new AsyncFieldResolver<object, DeletionStatus>(async (context) =>
+            {
+                var contentItemId = context.GetArgument<string>("contentItemId");
+
+                var contentManager = httpContextAccessor.HttpContext.RequestServices.GetRequiredService<IContentManager>();
```

Comment:
maybe better to use the field `_httpContextAccessor` if you initialize it before (not at the end), or just not declare it.